packages/rum/src/domain/segmentCollection/segmentCollection.ts

```diff
@@ -91,7 +91,7 @@ export function doStartSegmentCollection(
   const { unsubscribe: unsubscribePageExited } = lifeCycle.subscribe(
     LifeCycleEventType.PAGE_EXITED,
     (pageExitEvent) => {
-      flushSegment(pageExitEvent.isUnloading ? 'before_unload' : 'visibility_hidden')
+      flushSegment(pageExitEvent.reason === PageExitReason.UNLOADING ? 'before_unload' : 'visibility_hidden')
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** it ‏could be nice to use page exit reason as a creation reason
```ts
flushSegment(pageExitEvent.reason)
```